presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -532,6 +531,8 @@ public void testArrayContains()
         assertFunction("CONTAINS(ARRAY [2.2, 1.1], 0000000000001.100)", BOOLEAN, true);
         assertFunction("CONTAINS(ARRAY [2.2, 001.20], 1.2)", BOOLEAN, true);
         assertFunction("CONTAINS(ARRAY [ARRAY [1.1, 2.2], ARRAY [3.3, 4.3]], ARRAY [3.3, 4.300])", BOOLEAN, true);
+        assertFunction("CONTAINS(ARRAY [ARRAY [1.1, 2.2], ARRAY [3.3, 4.3]], ARRAY [1.1, null])", BOOLEAN, null);
+        assertFunction("CONTAINS(ARRAY [ARRAY [1.1, 2.2], ARRAY [3.3, 4.3]], ARRAY [1.3, null])", BOOLEAN, false);
```

Comment:
What about `CONTAINS(ARRAY [ARRAY [1.1, null], ARRAY [3.3, 4.3]], ARRAY [1.1, null])`?

We need to decide on the semantics.

Now that we specified array equal semantics for null elements in structural types, I think the semantics here need to be specified. Previously, we delayed making a decision on these issues by simply failing these queries. But now that these queries will return a result, we have make a choice now. We want to avoid having to change the semantics in the future.

Reply:
Currently the semantic is `CONTAINS(ARRAY [NULL, 2, 3], NULL) => NULL`.

I'm not changing the semantics. I'm following the existing semantics. Current `CONTAINS` semantic are the same as for `IN` operator. Mean that the `EQUALS` operator is used for equality. And CONTAINS is equivalent to `VALUE = ARRAY[0] OR ... OR VALUE = ARRAY[N]` applying the next conditions `TRUE OR FALSE OR NULL => TRUE` and `FALSE OR NULL => NULL`

Reply:
> What about CONTAINS(ARRAY [ARRAY [1.1, null], ARRAY [3.3, 4.3]], ARRAY [1.1, null])?

Added

Reply:
Whenever we decided to change the semantic to use `IS DISTINCT` operator - we can do that for both, for complex and simple types.

@martint

Reply:
I'm wondering if it is good enough to fail whenever the `EQUALS` operator returns `TRUE`. For example `ARRAY[1, NULL] = ARRAY[2, NULL] => FALSE`, and that already implies some semantics.

We can also check every element if it is indeterminate. But this is not free.

Reply:
Based on the discussion offline decided to go with the next approach:

```
If it returns TRUE - that the objects are equal for sure. If the equals operator returns FALSE - those are different for sure. Not matter what semantic you chose - the TRUE and FALSE answers will always remain the same.
So we can bail if the EQUALS operator returns NULL, saying that indeterminate comparisons are not supported.
```

Reply:
In other words the method will fail if the EQUALS operator returns `NULL`